src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs

```diff
@@ -246,8 +240,14 @@ private string ModifyANCMPathInConfig(string replaceFlag, string dllName, string
 
         private string GetIISExpressPath()
         {
+            var programFiles = "Program Files";
+            if (DotNetCommands.IsRunningX86OnX64(DeploymentParameters.RuntimeArchitecture))
+            {
+                programFiles = "Program Files (x86)";
+            }
+
             // Get path to program files
-            var iisExpressPath = Path.Combine(Environment.GetEnvironmentVariable("SystemDrive") + "\\", "Program Files", "IIS Express", "iisexpress.exe");
+            var iisExpressPath = Path.Combine(Environment.GetEnvironmentVariable("SystemDrive") + "\\", programFiles, "IIS Express", "iisexpress.exe");
```

Comment:
Can we use environment variable here?

Reply:
No, the environment variable changes based on the current process architecture, but we're trying to launch a process with a different arch.